playwright/_impl/_set_input_files_helpers.py

```diff
@@ -0,0 +1,80 @@
+import base64
+import os
+import shutil
+from pathlib import Path
+from posixpath import basename
+from typing import TYPE_CHECKING, List, Union
+
+from playwright._impl._helper import Error, async_readfile
+from playwright._impl._writable_stream import WritableStream
+
+if TYPE_CHECKING:  # pragma: no cover
+    from playwright._impl._browser_context import BrowserContext
+
+from playwright._impl._api_structures import FilePayload
+
+SIZE_LIMIT_IN_BYTES = 50 * 1024 * 1024
+
+
+async def convert_input_files(
+    files: Union[str, Path, FilePayload, List[Union[str, Path]], List[FilePayload]],
+    context: "BrowserContext",
+):
```

Comment:
lets define an intermediate return structure like we have upstream, so we have some types which we can use.
```suggestion
) -> InputFilesList:
```